pkg/client/state_changes_test.go

```diff
@@ -0,0 +1,194 @@
+
+	expectedStateChanges := StateChanges{
+		Data: &DataEntries{
+			&proto.StringDataEntry{
+				Key:   "keyHello",
+				Value: "hello",
+			},
+		},
+		Transfers: []*TransferAction{
+			{
+				Address: expectedTransferAddr,
+				Asset:   proto.NewOptionalAssetWaves(),
+				Amount:  900000,
+			},
+		},
+		Issues: []*IssueAction{
+			{
+				AssetID:        expectedAssetId,
+				Name:           "CatCoin",
+				Description:    "Cats are the best",
+				Decimals:       6,
+				Reissuable:     true,
+				CompiledScript: expectedScript,
+			},
+		},
+		Reissues: []*ReissueAction{
+			{
+				AssetID:    expectedAssetId,
+				Reissuable: false,
+				Quantity:   10000,
+			},
+		},
+		Burns: []*BurnAction{
+			{
+				AssetID:  expectedAssetId,
+				Quantity: 10000,
+			},
+		},
+		SponsorFees: []*SponsorFeeAction{
+			{
+				AssetID:              expectedAssetId,
+				MinSponsoredAssetFee: 100,
+			},
+		},
+		Leases: []*LeaseAction{
+			{
+				ID:                  expectedLeaseID,
+				OriginTransactionId: expectedLeaseID,
+				Sender:              expectedLeaseSender,
+				Recipient:           expectedLeaseRecipient,
+				Amount:              100000000,
+				Height:              10596,
+				Status:              LeaseCanceledStatus,
+			},
+		},
+		LeaseCancel: []*LeaseCancelAction{
+			{
+				LeaseID: expectedLeaseID,
+			},
+		},
+		Invokes: []*InvokeAction{
+			{
+				DApp: expectedInvokeDApp,
+				Call: proto.FunctionCall{
+					Name: "printNumber",
+					Arguments: proto.Arguments{
+						proto.NewIntegerArgument(1000),
+					},
+				},
+				Payments: []*proto.ScriptPayment{
+					{
+						Asset:  *proto.NewOptionalAssetFromDigest(expectedAssetId),
+						Amount: 100000,
+					},
+				},
+			},
+		},
+	}
+
+	var actualChanges StateChanges
+	require.NoError(t, json.Unmarshal([]byte(jsonSrc), &actualChanges), "unmarshal StateChanges error")
+	require.True(t, reflect.DeepEqual(expectedStateChanges.Data, actualChanges.Data), "non equal 'data' field")
+	require.True(t, reflect.DeepEqual(expectedStateChanges.Transfers, actualChanges.Transfers), "non equal 'transfers' field")
+	require.True(t, reflect.DeepEqual(expectedStateChanges.Issues, actualChanges.Issues), "non equal 'issues' field")
+	require.True(t, reflect.DeepEqual(expectedStateChanges.Reissues, actualChanges.Reissues), "non equal 'reissues' field")
+	require.True(t, reflect.DeepEqual(expectedStateChanges.Burns, actualChanges.Burns), "non equal 'burns' field")
+	require.True(t, reflect.DeepEqual(expectedStateChanges.SponsorFees, actualChanges.SponsorFees), "non equal 'sponsorFees' field")
+	require.True(t, reflect.DeepEqual(expectedStateChanges.Leases, actualChanges.Leases), "non equal 'leases' field")
+	require.True(t, reflect.DeepEqual(expectedStateChanges.LeaseCancel, actualChanges.LeaseCancel), "non equal 'leaseCancel' field")
+	require.True(t, reflect.DeepEqual(expectedStateChanges.Invokes, actualChanges.Invokes), "non equal 'invokes' field")
```

Comment:
> Under the hood require.Equal uses reflect.DeepEqual in some cases.

Didn't know, thanks. Ok, then we can use `require.Equal` instead of `require.True(reflect.DeepEqual())`

> These lines can be replaced by `require.Equal(t, expectedStateChanges, actualChanges)`

Agree. Done.